marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/RowBatcherImpl.java

```diff
@@ -0,0 +1,312 @@
+
+        return this;
+    }
+
+    @Override
+    public RowBatchSuccessListener[] getSuccessListeners() {
+        return new RowBatchSuccessListener[0];
+    }
+
+    @Override
+    public RowBatchFailureListener[] getFailureListeners() {
+        return failureListeners.toArray(new RowBatchFailureListener[failureListeners.size()]);
+    }
+
+    @Override
+    public void setSuccessListeners(RowBatchSuccessListener[] listeners) {
+        this.listeners = listeners;
+    }
+
+    @Override
+    public void setFailureListeners(RowBatchFailureListener... listeners) {
+
+    }
+
+    @Override
+    public boolean awaitCompletion() {
+        try {
+            return awaitCompletion(Long.MAX_VALUE, TimeUnit.DAYS);
+        } catch(InterruptedException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean awaitCompletion(long timeout, TimeUnit unit) throws InterruptedException {
+        if ( threadPool == null ) {
+            throw new IllegalStateException("Job not started.");
+        }
+        return threadPool.awaitTermination(timeout, unit);
+    }
+
+    @Override
+    public void retry(RowBatchRequestEvent event) {
+        if ( isStopped() == true ) {
+            logger.warn("Job is now stopped, aborting the retry");
+            return;
+        }
+        start(getJobTicket());
+    }
+
+    @Override
+    public void retryWithFailureListeners(RowBatchRequestEvent event) {
+        retry(event);
+    }
+
+    @Override
+    public JobTicket getJobTicket() {
+        return this.jobTicket;
+    }
+
+    @Override
+    public void stop() {
+        stopped.set(true);
+        if ( threadPool != null ) threadPool.shutdownNow();
+        jobEndTime = Calendar.getInstance();
+
+    }
+
+    @Override
+    public Calendar getJobStartTime() {
+        return this.jobStartTime;
+    }
+
+    @Override
+    public Calendar getJobEndTime() {
+        return this.jobEndTime;
+    }
+
+    @Override
+    public DatabaseClient getPrimaryClient() {
+        return null;
+    }
+
+    @Override
+    public void start(JobTicket ticket) {
+        if ( threadPool != null ) {
+            logger.warn("startJob called more than once");
+            return;
+        }
+        if ( getBatchSize() <= 0 ) {
+            withBatchSize(1);
+            logger.warn("batchSize should be 1 or greater--setting batchSize to 1");
+        }
+        jobTicket = ticket;
+
+        getRowCount();
+        jobStartTime = Calendar.getInstance();
+        started.set(true);
+    }
+
+
+    @Override
+    public boolean isStopped() {
+        return stopped.get();
+    }
+
+    @Override
+    public boolean isStarted() {
+        return started.get();
+    }
+
+    private void getRowCount(){
+
+        JsonNode jn = ((DatabaseClientImpl) client).getServices()
+                .getResource(null, "internal/viewinfo", null, params,
+                        new JacksonHandle())
+                .get();
```

Comment:
The value of rowCount used in the calculation on current line 248 is obtained by the internal/viewinfo request.

This method should get the value of the tableID and rowCount properties from the JsonNode and assign them to private fields of the RowBatcherImpl.